File: python/agentic_commerce_protocol/models/payment_method_card.py
# this file is @generated
import typing as t

from .card_funding_type import CardFundingType
from .card_number_type import CardNumberType
from .check_performed_type import CheckPerformedType
from .common import BaseModel


class PaymentMethodCard(BaseModel):
    type: str
    card_number_type: CardNumberType
    number: str
    exp_month: t.Optional[str] = None
    exp_year: t.Optional[str] = None
    name: t.Optional[str] = None
    cvc: t.Optional[str] = None
    cryptogram: t.Optional[str] = None
    eci_value: t.Optional[str] = None
    checks_performed: t.Optional[t.List[CheckPerformedType]] = None
    iin: t.Optional[str] = None
    display_card_funding_type: CardFundingType
    display_wallet_type: t.Optional[str] = None
    display_brand: t.Optional[str] = None
    display_last4: t.Optional[str] = None
    metadata: t.Dict[str, str]

File: kotlin/src/main/kotlin/com/svix/agentic_commerce_protocol/PaymentMethodCard.kt
// This file is @generated
package com.svix.agentic_commerce_protocol

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class PaymentMethodCard(
    val type: String,
    @SerialName("card_number_type") val cardNumberType: CardNumberType,
    val number: String,
    @SerialName("exp_month") val expMonth: String? = null,
    @SerialName("exp_year") val expYear: String? = null,
    val name: String? = null,
    val cvc: String? = null,
    val cryptogram: String? = null,
    @SerialName("eci_value") val eciValue: String? = null,
    @SerialName("checks_performed") val checksPerformed: List<CheckPerformedType>? = null,
    val iin: String? = null,
    @SerialName("display_card_funding_type") val displayCardFundingType: CardFundingType,
    @SerialName("display_wallet_type") val displayWalletType: String? = null,
    @SerialName("display_brand") val displayBrand: String? = null,
    @SerialName("display_last4") val displayLast4: String? = null,
    val metadata: Map<String, String>,
)
